corehq/ex-submodules/auditcare/tests/test_gdpr_scrub_user_auditcare.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from django.test import TestCase
+from django.core import management
+from auditcare.utils.export import get_auditcare_docs_by_username
+from auditcare.models import NavigationEventAudit
+from dimagi.utils.couch.database import iter_bulk_delete
+
+
+class TestGDPRScrubUserAuditcare(TestCase):
+
+    def setUp(self):
+        NavigationEventAudit(user="test_user1", request_path="/fake/path/0").save()
+        NavigationEventAudit(user="test_user1", request_path="/fake/path/1").save()
+        NavigationEventAudit(user="test_user1", request_path="/fake/path/2").save()
+
+    def tearDown(self):
+        all_auditcare_ids = list(set([result["id"] for result in NavigationEventAudit.get_db().view(
+            "auditcare/urlpath_by_user_date",
+            reduce=False,
+        ).all()]))
+        iter_bulk_delete(NavigationEventAudit.get_db(), all_auditcare_ids)
+
+    def test_update_username_no_returned_docs(self):
+        management.call_command("gdpr_scrub_user_auditcare", "nonexistent_user")
+        num_redacted_users = 0
+        for _ in get_auditcare_docs_by_username("Redacted User (GDPR)"):
+            num_redacted_users += 1
+        self.assertEqual(num_redacted_users, 0)
+        num_original_users = 0
+        for _ in get_auditcare_docs_by_username("test_user1"):
+            num_original_users += 1
+        self.assertEqual(num_original_users, 3)
+
+    def test_update_username_returned_docs(self):
+        management.call_command("gdpr_scrub_user_auditcare", "test_user1")
+        num_redacted_users = 0
+        for _ in get_auditcare_docs_by_username("Redacted User (GDPR)"):
+            num_redacted_users += 1
+        self.assertEqual(num_redacted_users, 3)
+        num_original_users = 0
+        for _ in get_auditcare_docs_by_username("test_user1"):
```

Comment:
Is there a reason you used this function rather than `get_num_auditcare_events_by_username` for these tests?  It looks like that one isn't used anywhere, but does essentially this.

Reply:
Oh, true. I think the solution would actually be to just delete `get_num_auditcare_events_by_username`, because otherwise the function that actually gets used (`get_auditcare_docs_by_username`) never actually gets tested. thoughts?

Reply:
(Made a PR just removing it if you agree with that solution)
https://github.com/dimagi/commcare-hq/pull/20034

Reply:
Actually I don't think `get_auditcare_docs_by_username` is used anywhere other than these tests.  The management command references `navigation_event_ids_by_user`.

Reply:
I see what you mean! I updated this to now test the right thing:

https://github.com/dimagi/commcare-hq/pull/20034